resolver.go

```diff
@@ -101,8 +101,8 @@ func (r *Resolver) LookupTXT(ctx context.Context, domain string) ([]string, erro
 }
 
 func (r *Resolver) getCachedIPAddr(domain string) ([]net.IPAddr, bool) {
-	r.RLock()
-	defer r.RUnlock()
+	r.mx.Lock()
```

Comment:
Switching away from RW mutexes may slow things down here. Given that we're caching against what would be a network request waiting on a mutex doesn't feel too bad, and if this becomes an issue we can always come back and optimize it.

Reply:
let's add a follow up issue? It shouldn't be hard to do, just queue the entries for cleanup and do that with write lock.

Reply:
https://github.com/libp2p/go-doh-resolver/issues/5